### PR TITLE
Scripts/DK: Attempt to fix a Death Pact exploit

### PR DESCRIPTION
**Pull request by CraftedRO**
**Changes proposed:**

-  Attempt to fix an exploit where Death Pact does heal (it is usable) while the ghoul is under the effect of cyclone and doesn't sacrifice him.

**Issues addressed:**

Closes https://github.com/TrinityCore/TrinityCore/issues/28874

**Tests performed:**

<details>
  <summary>Before:</summary>
  
https://user-images.githubusercontent.com/24683355/227716056-905264ac-dba7-42ae-bd4f-dd6f9f0e1862.mp4
  
</details>
<details>
  <summary>After:</summary>
  

https://github.com/TrinityCore/TrinityCore/assets/24683355/8dd2d76f-9a0a-4e94-98c7-b034da767353

  
</details>

Does it build, tested in-game as you can see above:

**Known issues and TODO list:** (add/remove lines as needed)

### CODE REVIEW

File: src/server/scripts/Spells/spell_dk.cpp [hunk 1]
```diff
@@ -762,13 +762,26 @@ class spell_dk_death_pact : public SpellScript
...
+            {
                 if (Creature* undeadPet = (*itr)->ToCreature())
+                {
                     if (undeadPet->IsAlive() &&
                         undeadPet->GetOwnerGUID() == player->GetGUID() &&
                         undeadPet->GetCreatureType() == CREATURE_TYPE_UNDEAD &&
                         undeadPet->IsWithinDist(player, 100.0f, false))
+                    {
+                        if (undeadPet->HasAura(33786) || undeadPet->HasAura(65859))
+                        {
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_dk.cpp:776]: remove brackets (single line if)

File: src/server/scripts/Spells/spell_dk.cpp [hunk 2]
```diff
@@ -762,13 +762,26 @@ class spell_dk_death_pact : public SpellScript
...
             for (Unit::ControlList::const_iterator itr = player->m_Controlled.begin(); itr != player->m_Controlled.end(); ++itr)
+            {
                 if (Creature* undeadPet = (*itr)->ToCreature())
+                {
                     if (undeadPet->IsAlive() &&
                         undeadPet->GetOwnerGUID() == player->GetGUID() &&
                         undeadPet->GetCreatureType() == CREATURE_TYPE_UNDEAD &&
                         undeadPet->IsWithinDist(player, 100.0f, false))
+                    {
+                        if (undeadPet->HasAura(33786) || undeadPet->HasAura(65859))
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_dk.cpp:775]: Don't check by spell id, use `IsImmunedToSpell`

**Review by Shauren — CHANGES_REQUESTED** [review]

File: src/server/scripts/Spells/spell_dk.cpp (hunk 2 above)
Comment (CraftedRO) [reply src/server/scripts/Spells/spell_dk.cpp:775]: well I tried some things with IsImmunedToSpell but didn't managed to compile, either I'm using wrong arguments either I shoud declare them first

Comment (CraftedRO) [reply src/server/scripts/Spells/spell_dk.cpp:775]: I tried many combinations with IsImmunedToSpell but it doesn't accept my arguments and I didn't quite understood how this works, also did not found any valid example with this function used elsewhere.

**Pull request merged**